tree/ntuple/v7/src/RNTuple.cxx

```diff
@@ -295,6 +295,12 @@ ROOT::Experimental::RNTupleWriter::~RNTupleWriter()
       CommitCluster(true /* commitClusterGroup */);
       fSink->CommitDataset();
    } catch (const RException &err) {
```

Comment:
Is `CommitCluster` doing `fZipTasks->Wait();` internally? Hard for me to find out :-)

Reply:
The `RNTupleWriter::CommitCluster()` call, at some point ends up calling `RPageSink::CommitCluster()`, that when using parallel page compression in turn calls [`RPageSinkBuf::CommitClusterImpl()`](https://github.com/root-project/root/blob/master/tree/ntuple/v7/src/RPageSinkBuf.cxx#L129) (which then waits for all tasks to finish).

However, if the "small clusters" condition is not met (see [here](https://github.com/root-project/root/blob/master/tree/ntuple/v7/src/RNTuple.cxx#L339), `RPageSinkBuf::CommitClusterImpl()` is never called.  Consequently, when the `RNTupleWriter` instance is destroyed, there might still be unfinished tasks.